### PR TITLE
Fix buildsystem and tests

### PR DESCRIPTION
**Pull request by wjwwood**
I have several fixes here to both the build system and unit tests.

Also I have fixes for pcl-1.7 related changes and pcl_ros changes.

### CODE REVIEW

**Comment by hershwg** [comment]
Wow, thanks for all this.

It's a shame that catkin_download_test_data needs to be told the
destination for the file every time. That really bloats up the file with a
lot of repetition.

File: amcl/test/basic_localization.py [hunk 1]
```diff
@@ -16,67 +12,68 @@
 from tf.transformations import euler_from_quaternion
 from std_srvs.srv import Empty
 
+
 class TestBasicLocalization(unittest.TestCase):
+    def setUp(self):
```
Comment (hershwg) [inline amcl/test/basic_localization.py:17]: Why did this entire file need to be re-indented like this?

**Comment by hershwg** [comment]
In general I prefer if format changes are saved up for separate pull requests (or at least separate commits), so it is easy for me to look at the substantive changes and think about them.  As is, this pull request seems to be about 50% formatting changes and 50% substantive changes.

**Comment by hershwg** [comment]
I'm waiting to merge this on the release of debians for pcl_ros 1.1.0 and pcl_conversions.  We are using hydro-devel branch of navigation in our project and don't want to have to compile pcl_ros etc from source.

File: amcl/test/basic_localization.py (hunk 1 above)
Comment (wjwwood) [inline amcl/test/basic_localization.py:17]: There were different indentions in the file, which is bad for python, so I reformatted it to four spaces (PEP8).

**Comment by wjwwood** [comment]
> It's a shame that catkin_download_test_data needs to be told the
> destination for the file every time. That really bloats up the file with a
> lot of repetition.

That's true, it has a default location, but it the default wouldn't work as well in this case.

> In general I prefer if format changes are saved up for separate pull requests (or at least separate commits), so it is easy for me to look at the substantive changes and think about them. As is, this pull request seems to be about 50% formatting changes and 50% substantive changes.

Sorry I started out like that, but things got mixed up, I'll try to split things up.

> I'm waiting to merge this on the release of debians for pcl_ros 1.1.0 and pcl_conversions. We are using hydro-devel branch of navigation in our project and don't want to have to compile pcl_ros etc from source.

I released them last night, but there appears to be issues on the farm, I'll be looking into that today.

**Comment by hershwg** [comment]
The formatting thing is just a preference of mine, it's not something you
need to rework the current pull request for.  I don't want to make more
work for you, I was just whining.

**Comment by wjwwood** [comment]
@hershwg no I totally agree, I planned to do a layered pull request, but I just wanted to get the changes commit and proposed last night. I really would like this stuff to get reviewed, because I may have made mistakes, and that is all the easier more accurate with more granular changes.

**Comment by wjwwood** [comment]
@hershwg ok, this should be much easier to review now. And if you don't like or agree with some of the style changes I can cherry pick them out.

**Comment by wjwwood** [comment]
I just added a commit which removes the rosbuild only `mainpage.dox` files.

**Comment by wjwwood** [comment]
I am going to fixup the `costmap_2d` tests as well.

**Comment by wjwwood** [comment]
I added some more commits which fix testing problems in `amcl`, `costmap_2d`, and `navfn`.

I can run all of the tests for navigation (and they pass):

```
% catkin_make run_tests
Base path: /Users/william/nav_ws/nav_only_ws
Source space: /Users/william/nav_ws/nav_only_ws/src
Build space: /Users/william/nav_ws/nav_only_ws/build
Devel space: /Users/william/nav_ws/nav_only_ws/devel
Install space: /Users/william/nav_ws/nav_only_ws/install
####
#### Running command: "make cmake_check_build_system" in "/Users/william/nav_ws/nav_only_ws/build"
####
Forcing "-j1" for running unit tests.
####
#### Running command: "make run_tests -j1" in "/Users/william/nav_ws/nav_only_ws/build"
####
[  0%] Built target geometry_msgs_generate_messages_cpp
[  0%] Built target std_msgs_generate_messages_cpp
[  0%] Built target costmap_2d_generate_messages_cpp
[  0%] Built target costmap_2d_gencpp
[  2%] Built target voxel_grid
[ 13%] Built target costmap_2d_gencfg
[ 31%] Built target costmap_2d
[ 36%] Built target navfn
[ 42%] Built target path_calc_test
[ 44%] Built target image_loader
[ 47%] Built target utest
[ 50%] Built target voxel_grid_tests
[ 52%] Built target base_local_planner_generate_messages_cpp
[ 52%] Built target base_local_planner_gencpp
[ 52%] Built target base_local_planner_gencfg
[ 86%] Built target base_local_planner
[ 89%] Built target trajectory_planner_ros
[100%] Built target base_local_planner_utest
[100%] Built target line_iterator
[100%] Built target tests
Scanning dependencies of target clean_test_results
[100%] Built target clean_test_results
Scanning dependencies of target _run_tests_navfn_gtest_path_calc_test
-- run_tests.py: execute commands
  /Users/william/nav_ws/nav_only_ws/devel/lib/navfn/path_calc_test --gtest_output=xml:/Users/william/nav_ws/nav_only_ws/build/test_results/navfn/gtest-path_calc_test.xml
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from PathCalc
[ RUN      ] PathCalc.oscillate_in_pinch_point
readPGM() Reading costmap file /Users/william/nav_ws/nav_only_ws/src/navigation/navfn/test/willow_costmap.pgm
readPGM() Size: 1132 x 1217
readPGM() Found 1228386 obstacle cells, 0 free cells, 0 unknown cells
[       OK ] PathCalc.oscillate_in_pinch_point (80 ms)
[ RUN      ] PathCalc.easy_nav_should_always_work
readPGM() Reading costmap file /Users/william/nav_ws/nav_only_ws/src/navigation/navfn/test/willow_costmap.pgm
readPGM() Size: 1132 x 1217
readPGM() Found 1228386 obstacle cells, 0 free cells, 0 unknown cells
[       OK ] PathCalc.easy_nav_should_always_work (41 ms)
[----------] 2 tests from PathCalc (122 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (122 ms total)
[  PASSED  ] 2 tests.
-- run_tests.py: verify result "/Users/william/nav_ws/nav_only_ws/build/test_results/navfn/gtest-path_calc_test.xml"

============ SOME CONTENT REMOVED ============

[----------] Global test environment tear-down
[==========] 25 tests from 4 test cases ran. (1 ms total)
[  PASSED  ] 25 tests.
-- run_tests.py: verify result "/Users/william/nav_ws/nav_only_ws/build/test_results/base_local_planner/gtest-base_local_planner_utest.xml"
[100%] Built target _run_tests_base_local_planner_gtest_base_local_planner_utest
Scanning dependencies of target _run_tests_base_local_planner_gtest
[100%] Built target _run_tests_base_local_planner_gtest
Scanning dependencies of target _run_tests_base_local_planner
[100%] Built target _run_tests_base_local_planner
Scanning dependencies of target run_tests
[100%] Built target run_tests

% catkin_test_results build/test_results
Summary: 37 tests, 0 errors, 0 failures
```

**Comment by wjwwood** [comment]
@hershwg These are all of the changes I have for this pull request. I might open another one later with more style changes for the `CMakeLists.txt`.

Let me know if you need to discuss any of these changes. `pcl` and `pcl_conversions` are built on the farm, I am fielding another fix for `pcl_ros` right now which should fix it too.

**Comment by hershwg** [comment]
Wow, what a lot of work!  Thanks William.

**Pull request merged**